lib/fastlane/plugin/ddg_apple_automation/actions/asana_log_message_action.rb

```diff
@@ -0,0 +1,96 @@
+require "fastlane/action"
+require "fastlane_core/configuration/config_item"
+require "asana"
+require_relative "../helper/ddg_apple_automation_helper"
+require_relative "asana_add_comment_action"
+require_relative "asana_get_release_automation_subtask_id_action"
+require_relative "asana_get_user_id_for_github_handle_action"
+require_relative "asana_extract_task_id_action"
+require_relative "asana_extract_task_assignee_action"
+
+module Fastlane
+  module Actions
+    class AsanaLogMessageAction < Action
+      def self.run(params)
+        token = params[:asana_access_token]
+        task_url = params[:task_url]
+        template_name = params[:template_name]
+        comment = params[:comment]
+        is_scheduled_release = params[:is_scheduled_release] || false
+        github_handle = params[:github_handle]
+
+        automation_subtask_id = AsanaGetReleaseAutomationSubtaskIdAction.run(task_url: task_url, asana_access_token: token)
+
+        if is_scheduled_release
+          task_id = AsanaExtractTaskIdAction.run(task_url: task_url)
+          assignee_id = AsanaExtractTaskAssigneeAction.run(task_id: task_id, asana_access_token: token)
+        else
+          if github_handle.to_s.empty?
+            UI.user_error!("Github handle cannot be empty for manual release")
+            return
+          end
+          assignee_id = AsanaGetUserIdForGithubHandleAction.run(github_handle: github_handle, asana_access_token: token)
+        end
+
+        asana_client = Asana::Client.new do |c|
+          c.authentication(:access_token, token)
+        end
+
+        begin
+          asana_client.tasks.add_followers_for_task(task_gid: automation_subtask_id, followers: [assignee_id])
+        rescue StandardError => e
+          UI.user_error!("Failed to add a collaborator to the release task: #{e}")
```

Comment:
Technically it's not the release task that the user is added to, but the 'Automation' subtask of a release task. How about the following? To allow easier debugging, perhaps.

```suggestion
          UI.user_error!("Failed to add user #{assignee_id} as collaborator on task #{automation_subtask_id}: #{e}")
```